typescript/sdk/src/gas/HyperlaneIgpDeployer.ts

```diff
@@ -126,84 +109,73 @@ export class HyperlaneIgpDeployer extends HyperlaneDeployer<
-        this.multiProvider.getSigner(chain),
-      );
-      if (configsToSet[gasOracle].length > 0) {
-        await this.runIfOwner(chain, gasOracleContract, async () => {
-          this.logger(
-            `Setting gas oracle on ${gasOracle} for ${configsToSet[
-              gasOracle
-            ].map((config) => config.remoteDomain)}`,
+
+    if (configsToSet.length > 0) {
+      await this.runIfOwner(chain, gasOracle, async () => {
+        configsToSet.forEach((v) => {
+          const exchangeRate = ethers.utils.formatUnits(
+            v.tokenExchangeRate as any,
+            TOKEN_EXCHANGE_RATE_EXPONENT,
           );
-          return this.multiProvider.handleTx(
-            chain,
-            gasOracleContract.setRemoteGasDataConfigs(
-              configsToSet[gasOracle],
-              this.multiProvider.getTransactionOverrides(chain),
-            ),
+          const gasPrice = ethers.utils.formatUnits(v.gasPrice as any, 'gwei');
+          this.logger(
+            `Updating ${v.remote.padEnd(15)}\t${exchangeRate.padEnd(
+              10,
+            )}\t${gasPrice} gwei`,
           );
         });
-      }
+        return this.multiProvider.handleTx(
+          chain,
+          gasOracle.setRemoteGasDataConfigs(
+            configsToSet,
+            this.multiProvider.getTransactionOverrides(chain),
+          ),
+        );
+      });
     }
+
+    return gasOracle;
   }
 
   async deployContracts(
     chain: ChainName,
-    config: IgpConfig & Partial<OracleConfig>,
+    config: IgpConfig,
   ): Promise<HyperlaneContracts<IgpFactories>> {
     // NB: To share ProxyAdmins with HyperlaneCore, ensure the ProxyAdmin
     // is loaded into the contract cache.
     const proxyAdmin = await this.deployContract(chain, 'proxyAdmin', []);
 
-    const storageGasOracle = await this.deployStorageGasOracle(chain);
+    const storageGasOracle = await this.deployStorageGasOracle(chain, config);
     const interchainGasPaymaster = await this.deployInterchainGasPaymaster(
       chain,
       proxyAdmin,
       storageGasOracle,
       config,
     );
 
-    // Configure storage gas oracle with remote gas data if provided
-    if (config.oracleConfig) {
-      await this.configureStorageGasOracle(
-        chain,
-        interchainGasPaymaster,
-        config.oracleConfig,
-      );
-    }
-
-    await this.transferOwnershipOfContracts(chain, config.owner, {
-      interchainGasPaymaster,
-    });
-
-    // Configure oracle key for StorageGasOracle separately to keep 'hot'
-    // for updating exchange rates regularly
-    await this.transferOwnershipOfContracts(chain, config.oracleKey, {
-      storageGasOracle,
-    });
-
-    return {
+    const contracts = {
       proxyAdmin,
       storageGasOracle,
       interchainGasPaymaster,
     };
+
+    const ownerConfig = {
+      ...config,
+      ownerOverrides: {
+        ...config.ownerOverrides,
+        storageGasOracle: config.oracleKey, // TODO: merge this
```

Comment:
why is the oracleKey (address) assigned as the storageGasOracle

Reply:
because the "oracleKey" permission is really just the owner of Ownable StorageGasOracle